ui/src/app/shared/utils/date/dateutils.spec.ts

```diff
@@ -37,7 +37,7 @@ describe('DateUtils', () => {
   });
 
   it('#isDateBefore - checks if given date is before date to be compared to', () => {
-    const date: Date = DateUtils.stringToDate('2023-01-01');
+    const date: Date = DateUtils.stringToDate('2023-01-01') as Date;
```

Comment:
This works. lines below it also follows similar pattern. a slight modification like below can solve 5 other strict errors.
isDateBefore(date: Date, compareDate: Date) -> isDateBefore(date: Date | null, compareDate: Date | null).

But it's upto you.

Reply:
I see. thank you for your feedback. This time I want to focus just adding types, So I will keep this line. But I will memorize this information in mind.